### ISSUE TITLE
Document `ninja reconfigure` in the online documentation

### ISSUE THREAD

**Issue by adrianjohnson** [mesonbuild/meson#2409]
After the build dir is created, I can change meson files or options and running ninja will reconfigure if required then build. My project checks that various options are compatible and prints a summary of the build configuration. I would like to be able to perform a reconfigure with out building so I can see the configuration summary before I decide if I want to build the project.

Even better would be the way cmake does it. I can change run "cmake .. -Dopt=var" in my build dir and it prints out my build summary. It seems to figure out how to only reconfigure what has changed instead of reconfiguring everything then prints out the summary. I assume all messages not inside a if statement are getting printed.

**Comment by QuLogic** [comment]
Does `ninja reconfigure` do what you want?

**Comment by adrianjohnson** [comment]
Yes that works. I didn't see that anywhere in the meson documentation. Maybe it could be added to http://mesonbuild.com/Running-Meson.html

it could still be improved. I see it is skipping the dependency checks, however it is stil running the compiler argument checks, header file checks, and function checks. Could meson cache the results of these checks?

**Comment by nirbheek** [comment]
> it could still be improved. I see it is skipping the dependency checks, however it is stil running the compiler argument checks, header file checks, and function checks. Could meson cache the results of these checks?

That feature is planned to be implemented for the 0.44.0 release.

**Comment by bredelings** [comment]
Yeah, we should document this.  The only thing I do more frequently than running ninja install is 
```
rm -rf gcc-8-debug
CXXFLAG=<stuff> CXX=g++-8 CC=gcc-8 meson ../master gcc-8-debug --buildtype=debug --prefix=$HOME/Devel/asdf/local
```
This takes enough effort that I ended up writing a script to regenerate meson build directories without essentially writing a small program on the command line each time the meson version is changed.

For autoconf, you can do `head config.log`, copy the configure command, and then `rm -rf` and reconfigure (though to do have to requote spaces in CXXFLAGS).
